Remove Firfox warning...

schnepe2 opened the pull request:
`Expected ‘none’, URL, or filter function but found ‘progid’.  Error in parsing value for ‘filter’.  Declaration dropped.` during startup.

schnepe2 committed:
Removes Firfox warning...

"Expected ‘none’, URL, or filter function but found ‘progid’.  Error in parsing value for ‘filter’.  Declaration dropped." during startup.

level420 committed:
Merge branch 'master' into remove-FF-warning

hkollmann reviewed (changes requested):
filter is supported since firefox 35.

http://www.w3schools.com/jsref/prop_style_filter.asp
https://developer.mozilla.org/de/docs/Web/CSS/filter

which firefox version are you using?

schnepe2 commented:
Currently 75.0 and 76.0 (windows)
That warning is showing up in FFs for quite a while though (at least since 57.0)

schnepe2 commented:
> filter is supported since firefox 35.

filter is supported, but not the construct that qooxdoo is using to test it!
As the warning says!
'progid: ....'

hkollmann reviewed (commented):
Then the test is wrong. With your change the answer of __isFilterSupported is false for FF. Which is not correct.
Can we change the test instead not using progid?

schnepe2 commented:
> Then the test is wrong. 
No it isn't! The method `__isFilterSupported` is only for checking some IE5.5 specific features.
See `getFilterGradient` and `getFilterTextShadow`
The name might not be perfect for that method however.
And documentation is also weak.

> With your change the answer of __isFilterSupported is false for FF. Which is not correct.
That result **is** correct for FF (see above)

> Can we change the test instead not using progid?
The method is only for testing special IE cases, so it's O.K. this way.

oetiker commented on framework/source/class/qx/bom/client/Css.js:
```diff
@@ -426,18 +426,20 @@ qx.Bootstrap.define("qx.bom.client.Css",
+      if (qx.core.Environment.get("browser.name") != "firefox")
       {
-        supported = true;
-      }
-      document.body.removeChild(el);
+        var value = "progid:" + filterClass + "(" + initParams + ");";
+        var el = document.createElement("div");
+        document.body.appendChild(el);
+        el.style.filter = value;
 
+        if (el.filters && el.filters.length > 0 &&
+          el.filters.item(filterClass).enabled == true)
+        {
+          supported = true;
+        }
+        document.body.removeChild(el);
+      }
```
I would suggest if we touch this, then to deprecate the functions and always return `false` 
see also https://developer.mozilla.org/en-US/docs/Web/CSS/-ms-filter

we are not talking about genuine filter here but a M$ hack not required since IE8!

doing the whole sing and dance just takes valuable time ... at execution time.

better still remove the entire function and add deprecation notices to the getFilterGradient and getFilterTextShadow  calls.

oetiker replied:
```suggestion
```

oetiker commented on framework/source/class/qx/bom/client/Css.js:
```diff
@@ -426,18 +426,20 @@ qx.Bootstrap.define("qx.bom.client.Css",
-
-      if (el.filters && el.filters.length > 0 &&
-        el.filters.item(filterClass).enabled == true)
+      if (qx.core.Environment.get("browser.name") != "firefox")
```
```suggestion
```

hkollmann commented:
I would agree to @oetiker. IE 5.5 is dead. Return false and add an deprecated comment.

hkollmann committed:
Merge branch 'master' into remove-FF-warning

hkollmann commented:
@oetiker :  Removing the check and return always false leeds to a breaking change in 
https://github.com/qooxdoo/qooxdoo/blob/master/framework/source/class/qx/ui/decoration/MLinearBackgroundGradient.js#L155.

css.gradient.filter is set through
 qx.core.Environment.add("css.gradient.filter", statics.getFilterGradient);  

So i suggest to accept  the PR

hkollmann committed:
Merge branch 'master' into remove-FF-warning

oetiker commented:
@hkollmann the change as it stands makes no sense ... why test for obscrure things present only in browsers we do not support and then go ahead and exclude firefox just because it complains ... 

I can't see why this would break anything ...

``` javascript
__isFilterSupported : function(filterClass, initParams) {
  return false;
},
```

hkollmann commented:
Then we can remove this too:
from https://github.com/qooxdoo/qooxdoo/blob/master/framework/source/class/qx/ui/decoration/MLinearBackgroundGradient.js#L155:

      var styleImpl = this.__styleLinearBackgroundGradientAccordingToSpec;
      if (qx.core.Environment.get("css.gradient.legacywebkit")) {
        styleImpl = this.__styleLinearBackgroundGradientForLegacyWebkit;
      } else if (qx.core.Environment.get("css.gradient.filter") &&
        !qx.core.Environment.get("css.gradient.linear") && qx.core.Environment.get("css.borderradius")) {
        styleImpl = this.__styleLinearBackgroundGradientWithCanvas;
      } else if (qx.core.Environment.get("css.gradient.filter") &&
        !qx.core.Environment.get("css.gradient.linear")) {
        styleImpl = this.__styleLinearBackgroundGradientWithMSFilter;
      }

css.gradient.filter depends on __isFilterSupported in statics.getFilterGradient)

This is the only place where the settings are used.

hkollmann committed:
Merge branch 'master' into remove-FF-warning

schnepe2 committed:
Complete removal of obsolete IE5.5 filter checks as proposed by @hkollmann

- Changes have been documented in CHANGELOG.md
- Trailng whitespaces (in `MLinearBackgroundGradient.js`) removed by editor. Functional changes only in `_styleLinearBackgroundGradient`.

schnepe2 commented:
@hkollmann : ...like this? If accepted I'll also add the changes to #9909

schnepe2 committed:
Update CHANGELOG.md

hkollmann reviewed (approved):
great!

oetiker reviewed (approved):
yessss :)
